src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -91,6 +107,161 @@ class LspConnectionManager {
+      clickableInvisibleDiv.style.left = (syncIcon as any).offsetLeft + 'px'
+      clickableInvisibleDiv.style.height =
+        ((syncIcon as any).offsetHeight || 0) + 'px'
+      clickableInvisibleDiv.style.width =
+        ((syncIcon as any).offsetWidth || 0) + 'px'
+
+      clickableInvisibleDiv.removeEventListener('click', () =>
+        this._toggleSchemaSync()
+      ) // may have existing
+      clickableInvisibleDiv.addEventListener('click', () =>
+        this._toggleSchemaSync()
+      )
+    }, 1000)
+  }
+
+  _initComposition(schema: SchemaSelection) {
+    if (!this._canInitComposition) {
+      return
+    }
+
+    // msg to LSP
+    const {bucket, measurement} = schema
+    const payload = {bucket: bucket?.name}
+    if (measurement) {
+      payload['measurement'] = measurement
+    }
+    // TODO: finish LSP update first
+    // this.inject(ExecuteCommand.CompositionInit, payload)
+
+    // msg to monaco-editor
+    this._setCompositionBlockStyle(
+      true,
+      schema.composition?.startLine || 1,
+      schema.composition?.startLine || 6
+    )
+
+    // handlers to trigger end composition
+    this._setCompositionToggle()
+    this._model.onDidChangeContent(e => {
+      const {changes} = e
+      changes.forEach(change => {
+        if (
+          change.range.startLineNumber >=
+            (schema.composition?.startLine || 1) &&
+          change.range.endLineNumber <= (schema.composition?.endLine || 6)
+        ) {
+          this._canInitComposition = false // TODO: eventually, will be based on LSP response
+          this._toggleSchemaSync()
+          return
+        }
+      })
+    })
+  }
+
+  _endComposition(schema: SchemaSelection) {
+    // msg to LSP
+    // TODO: is there any msg to the LSP on composition end?
+    // can we have it change the yield unique identifier?
+    // this.inject(ExecuteCommand.CompositionInit, {})
+
+    // msg to monaco-editor
+    this._setCompositionBlockStyle(
+      false,
+      schema.composition?.startLine || 1,
+      schema.composition?.startLine || 1
+    )
+  }
+
+  // hack for now. To trigger for example video.
+  private _turnOnWithFirstCall = true
+
+  onSchemaSessionChange(schema: SchemaSelection, sessionCb) {
+    this._callbackSetSession = sessionCb
+    const previousState = this._session
+    this._session = {...schema}
+
+    if (!previousState?.synced && schema.synced) {
+      return this._initComposition(schema)
+    }
+    if (previousState?.synced && !schema.synced) {
+      return this._endComposition(schema)
+    }
+    // TODO: all other updates
+    // will use this.inject() --> send to LSP
+
+    // hack for now. To trigger for example video.
+    if (this._turnOnWithFirstCall) {
+      this._turnOnWithFirstCall = false
+      setTimeout(() => {
+        console.error('MOCK SCHEMA INJECTION FROM USER')
+        this._toggleSchemaSync()
+      }, 5000)
+    }
+  }
+
+  onLspCompositionChange(_jsonrpcMiddlewareResponse: unknown) {
```

Comment:
this is the method, which will be subscribing to all LSP outbound communications (after middleware is in place).